### PR TITLE
Fix script logic of cults of tibia quest (The Sinister Hermit)

### PR DESCRIPTION
**Pull request by dudantas**
Working with: https://github.com/opentibiabr/canary/pull/389/files

The real name of the monster was replaced by the name of the look, generating unexpected behavior and in a very specific scenario a crash (when two different monsters had the same "name")
Some improvements related to the creation of monsters and the verification of the monster name in getMonsterType, also preventing any unexpected behavior

### CODE REVIEW

**Review by majestyotbr — APPROVED** [review]

File: data/scripts/creaturescripts/quests/cults_of_tibia/evaporate.lua [hunk 1]
```diff
@@ -17,7 +17,7 @@ function spellCallback(param)
 	if tile then
 		if tile:getTopCreature() and tile:getTopCreature():isMonster() then
 			if tile:getTopCreature():getName():lower() == "leiden" then
-				tile:getTopCreature():registerEvent("spawnBoss")
+				tile:getTopCreature():registerEvent("SpawnBoss")
```
Comment (guispiller) [inline data/scripts/creaturescripts/quests/cults_of_tibia/evaporate.lua]: To avoid conflicts, you can revert this change, I changed this file and fix that script on #604

**Review by majestyotbr — APPROVED** [review]

**Review by guispiller — APPROVED** [review]
